### PR TITLE
feat: install pieces across platform

### PR DESCRIPTION
**Pull request by abuaboud**
## What does this PR do?

Fixes # (issue)

### CODE REVIEW

File: packages/backend/project.json [hunk 1]
```diff
@@ -64,14 +64,13 @@
         "jestConfig": "packages/backend/jest.config.ts",
         "passWithNoTests": false,
         "bail": true,
-        "testPathPattern": ["packages/backend/test/integration/cloud"]
+        "testPathPattern": ["packages/backend/test/integration/cloud/piece-metadata/piece-metadata.test.ts"]
```
Comment (khaledmashaly) [inline packages/backend/project.json]: Please revert changes to the testing configuration.
(blocking)

File: packages/backend/src/app/ee/pieces/platform-piece-module.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            ownerId: userId,
+        })
+        if (platform?.id !== platformId) {
+            throw new ActivepiecesError({
+                code: ErrorCode.AUTHORIZATION,
+                params: {},
+            })
+        }
+    }
+}
```
Comment (khaledmashaly) [inline packages/backend/src/app/ee/pieces/platform-piece-module.ts:55]: BTW, there's already a `checkUserIsOwner` method in the `platformService`!

(blocking)

File: packages/backend/src/app/ee/pieces/platform-piece-module.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        },
+    }, async (req): Promise<PieceMetadataModel> => {
+        const { platformId } = req.body
+        const { packageType, pieceName, pieceVersion, pieceArchive } = req.body
+        const { projectId } = req.principal
+        await assertUserIsPlatformOwner({
+            platformId,
+            userId: req.principal.id,
+        })
+        return await pieceService.add({
```
Comment (khaledmashaly) [inline packages/backend/src/app/ee/pieces/platform-piece-module.ts]: is it `pieceMetadataService#create` or `pieceService.add`?
(blocking)

File: packages/backend/src/app/pieces/piece-metadata-service/db-piece-metadata-service.ts [hunk 4]
```diff
@@ -148,6 +150,55 @@ export const DbPieceMetadataService = (): PieceMetadataService => {
...
+    pieceType: Equal(PieceType.CUSTOM),
+})
+
+const applyVersionFilter = (filters: Record<string, unknown>[], version: string): Record<string, unknown>[] => {
+    return filters.map(filter => ({
+        ...filter,
+        version: findSearchOperation(version),
+    }))
+}
+
```
Comment (khaledmashaly) [inline packages/backend/src/app/pieces/piece-metadata-service/db-piece-metadata-service.ts:201]: wooow :joy:

**Review by khaledmashaly — CHANGES_REQUESTED** [review]

File: packages/backend/src/app/ee/pieces/platform-piece-module.ts (hunk 3 above)
Comment (abuaboud) [reply packages/backend/src/app/ee/pieces/platform-piece-module.ts]: pieceService#add similar to community piece module

Comment (abuaboud) [reply packages/backend/src/app/ee/pieces/platform-piece-module.ts]: Renamed to Install Piece

**Pull request merged**